### PR TITLE
Decouple swaps metric logic from `TransactionController`

### PR DESCRIPTION
**Pull request by OGPoyraz**
## **Description**
With ongoing effort of `TransactionController` alignment between mobile and extension, this PR aims to decouple swaps metric logic from `TransactionController` and make them client specific.

Also removed `transaction-finalized` and put `transaction-confirmed` and `transaction-failed`. This should've been done before but I couldn't notice when I was working mainly focused on metric events.

## **Manual testing steps**

No functional changes. 

## **Related issues**

Fixes https://github.com/MetaMask/MetaMask-planning/issues/1053

## **Pre-merge author checklist**

- [X] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've clearly explained:
  - [X] What problem this PR is solving.
  - [X] How this problem was solved.
  - [X] How reviewers can test my changes.
- [X] I’ve indicated what issue this PR is linked to: Fixes #???
- [X] I’ve included tests if applicable.
- [X] I’ve documented any added code.
- [X] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: shared/constants/transaction.ts [hunk 1]
```diff
@@ -500,3 +506,19 @@ export enum TokenStandard {
...
+  added = 'transaction-added',
+  approved = 'transaction-approved',
+  confirmed = 'transaction-confirmed',
+  dropped = 'transaction-dropped',
+  failed = 'transaction-failed',
+  newSwap = 'transaction-new-swap',
+  newSwapApproval = 'transaction-new-swap-approval',
+  rejected = 'transaction-rejected',
+  submitted = 'transaction-submitted',
+  postTransactionBalanceUpdated = 'post-transaction-balance-updated',
```
Comment (matthewwalsh0) [inline shared/constants/transaction.ts:523]: Minor, but alphabetical?

File: app/scripts/metamask-controller.js [hunk 2]
```diff
@@ -4872,10 +4708,192 @@ export default class MetamaskController extends EventEmitter {
...
+      TransactionEvent.rejected,
+      handleTransactionRejected.bind(null, transactionMetricsRequest),
+    );
+    this.txController.on(
+      TransactionEvent.submitted,
+      handleTransactionSubmitted.bind(null, transactionMetricsRequest),
+    );
+
+    this.txController.on(`tx:status-update`, async (txId, status) => {
+      if (
```
Comment (matthewwalsh0) [inline app/scripts/metamask-controller.js:4811]: No action needed, but for audit sake, this huge listener callback isn't ideal and I've already split it out into multiple methods in the adoption branch.

**Review by matthewwalsh0 — DISMISSED** [review]

File: app/scripts/metamask-controller.js (hunk 2 above)
Comment (OGPoyraz) [reply app/scripts/metamask-controller.js:4811]: Exactly this doesn't look ideal, thanks for informing

File: shared/constants/transaction.ts (hunk 1 above)
Comment (OGPoyraz) [reply shared/constants/transaction.ts]: Fixed in https://github.com/MetaMask/metamask-extension/pull/21559/commits/7ef24787fd00bfd373c8209c9e6ea77bb2870a0f

File: app/scripts/metamask-controller.js [hunk 3]
```diff
@@ -4872,10 +4708,192 @@ export default class MetamaskController extends EventEmitter {
...
+  }
+
   //=============================================================================
   // CONFIG
   //=============================================================================
 
+  /**
+   * A method for setting TransactionController event listeners
+   */
+  _addTransactionControllerListeners() {
```
[the remark this answers is not in the document]
Comment (OGPoyraz) [reply app/scripts/metamask-controller.js:4714]: Relocated in https://github.com/MetaMask/metamask-extension/pull/21559/commits/4f44d52a5b938d8dd90d438bd699c216b55014a6

**Review by matthewwalsh0 — APPROVED** [review]

File: app/scripts/controllers/transactions/index.js [hunk 4]
```diff
@@ -705,7 +699,7 @@ export default class TransactionController extends EventEmitter {
       this.txStateManager.setTxStatusConfirmed(txId);
       this._markNonceDuplicatesDropped(txId);
 
-      this.emit('transaction-finalized', {
+      this.emit(TransactionEvent.confirmed, {
```
Comment (vinistevam) [inline app/scripts/controllers/transactions/index.js:702]: :heart:

**Review by vinistevam — APPROVED** [review]

**Pull request merged**